tests/mir-opt/tail_expr_drop_order_unwind.method_1.ElaborateDrops.after.panic-unwind.mir

```diff
@@ -78,106 +75,85 @@ fn method_1(_1: Guard) -> () {
     bb7: {
         backward incompatible drop(_2);
         backward incompatible drop(_5);
-        goto -> bb24;
+        goto -> bb21;
     }
 
     bb8: {
-        drop(_5) -> [return: bb9, unwind: bb16];
+        drop(_5) -> [return: bb9, unwind: bb13];
     }
 
     bb9: {
-        _11 = const false;
         StorageDead(_5);
         StorageDead(_4);
         _10 = const false;
         StorageDead(_2);
-        drop(_1) -> [return: bb10, unwind: bb17];
+        drop(_1) -> [return: bb10, unwind: bb14];
     }
 
     bb10: {
         return;
     }
 
     bb11 (cleanup): {
-        goto -> bb28;
+        goto -> bb25;
     }
 
     bb12 (cleanup): {
         drop(_5) -> [return: bb13, unwind terminate(cleanup)];
     }
 
     bb13 (cleanup): {
-        goto -> bb15;
+        drop(_1) -> [return: bb14, unwind terminate(cleanup)];
     }
 
     bb14 (cleanup): {
-        drop(_5) -> [return: bb15, unwind terminate(cleanup)];
+        resume;
     }
 
-    bb15 (cleanup): {
-        goto -> bb30;
+    bb15: {
+        goto -> bb8;
     }
 
     bb16 (cleanup): {
-        drop(_1) -> [return: bb17, unwind terminate(cleanup)];
+        goto -> bb12;
     }
 
     bb17 (cleanup): {
-        resume;
+        goto -> bb12;
     }
 
     bb18: {
-        goto -> bb8;
+        goto -> bb15;
     }
 
-    bb19 (cleanup): {
+    bb19: {
         goto -> bb15;
     }
 
     bb20 (cleanup): {
-        goto -> bb15;
+        goto -> bb12;
     }
 
     bb21: {
-        goto -> bb18;
+        _11 = discriminant(_2);
+        switchInt(move _11) -> [0: bb18, otherwise: bb19];
     }
 
-    bb22: {
-        goto -> bb18;
-    }
-
-    bb23 (cleanup): {
-        goto -> bb15;
-    }
-
-    bb24: {
+    bb22 (cleanup): {
         _12 = discriminant(_2);
-        switchInt(move _12) -> [0: bb21, otherwise: bb22];
+        switchInt(move _12) -> [0: bb16, otherwise: bb20];
     }
 
-    bb25 (cleanup): {
-        _13 = discriminant(_2);
-        switchInt(move _13) -> [0: bb19, otherwise: bb23];
-    }
-
-    bb26 (cleanup): {
+    bb23 (cleanup): {
         goto -> bb12;
     }
 
-    bb27 (cleanup): {
+    bb24 (cleanup): {
         goto -> bb12;
     }
 
-    bb28 (cleanup): {
-        _14 = discriminant(_2);
-        switchInt(move _14) -> [0: bb26, otherwise: bb27];
-    }
-
-    bb29 (cleanup): {
-        drop(_5) -> [return: bb16, unwind terminate(cleanup)];
```

Comment:
notably, this is the drop terminator that is bad. previously we were unwinding to  bb14, which would drop `_5`, then which branched to bb15 -> bb30 which tested a drop flag for `_5` (which was never set to false), which branched to bb29 where we would drop `_5` yet again.